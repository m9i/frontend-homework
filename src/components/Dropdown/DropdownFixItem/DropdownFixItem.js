import React from "react";

const DropdownFixItem = ({text, icon}) => {
    const className = `nav__list__item ${icon === 'logout' ? 'color-red' : ''}`;

    return <div className={className}>
        <i className="material-icons-outlined nav__list__item--icon">
            {icon}
        </i>
        {text}
    </div>
}

export default DropdownFixItem;
