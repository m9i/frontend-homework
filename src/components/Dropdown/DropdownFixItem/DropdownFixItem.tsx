import React from 'react';
import {IDropdownFixItemProps} from '../../../interfaces/IDropdownFixItemProps';

const DropdownFixItem: React.FC<IDropdownFixItemProps> = ({text, icon}: IDropdownFixItemProps) => {
    const className: string = `nav__list__item ${icon === 'logout' ? 'color-red' : ''}`;

    return (
        <div className={className}>
            <i className="material-icons-outlined nav__list__item--icon">{icon}</i>
            {text}
        </div>
    );
};

export default DropdownFixItem;
