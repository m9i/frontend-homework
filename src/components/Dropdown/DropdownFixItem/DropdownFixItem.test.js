import React from "react";
import DropdownFixItem from "./DropdownFixItem";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

describe("rendering <DropdownFixItem />", () => {
  it("renders DropdownFixItem component without crashing", () => {
    shallow(<DropdownFixItem />);
  });
});

describe("snapshots", () => {
  it("DropdownFixItem snapshot", () => {
    const dropDowmFixItemTree = shallow(<DropdownFixItem />);
    expect(toJson(dropDowmFixItemTree)).toMatchSnapshot();
  });
});
