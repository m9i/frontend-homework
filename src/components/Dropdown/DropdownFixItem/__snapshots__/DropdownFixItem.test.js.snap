// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots DropdownFixItem snapshot 1`] = `
<div
  className="nav__list__item "
>
  <i
    className="material-icons-outlined nav__list__item--icon"
  />
</div>
`;
